### ISSUE TITLE
Moved code UI is bit busy and full of lines and boxes

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#185957]
Loving the moved code detection but I think the UI can benefit from some polish. There are very many lines, boxes and intersections of things making it not fit so well into the general more flat look.

### Today (boxes, frames)

<img width="572" alt="Screenshot 2023-06-23 at 09 45 20" src="https://github.com/microsoft/vscode/assets/1794099/3bd61d24-8250-47f3-b68c-b1251d3a9a11">

### Idea (no extra box, a strong'ish connector)

Not shown because I lack skills is an idea to let the color of the vertical connecting line be a gradient from deleted to inserted color

<img width="572" alt="Screenshot 2023-06-23 at 09 45 20" src="https://github.com/microsoft/vscode/assets/1794099/411cac24-df43-4a07-9fda-1bb43200e6e5">

**Comment by jrieken** [comment]
cc @daviddossett for input/ideas and UI magic

**Comment by hediet** [comment]
Yep, I neglected the moved code feature a bit (even thinking of removing the menu entry, so that you can only enable it with diffEditor.experimental.detectMoves).

The extra box in your example is unneeded, but in principle one deletion can refer to multiple moves (not just one) - in that case you need boxes to see which part moved where. It is just that the diff algorithm cannot detect such cases yet (but the diff editor would show them correctly).

Current todos:
* [ ] Better handling of lines, they shouldn't overlap with line-number etc
   * [ ] Adding a gutter for that is probably the best idea as discussed in slack some time ago
* [ ] Better UX to align source and target code of a move (currently you just place the cursor inside it)

**Comment by jrieken** [comment]
> The extra box in your example is unneeded, but in principle one deletion can refer to multiple moves (not just one) - in that case you need boxes to see which part moved where.

Are making a case for even more boxes 🙈 ?

**Comment by hediet** [comment]
This is the result of the second iteration:

![Code_-_OSS_GW3McYmrkI](https://github.com/microsoft/vscode/assets/2931520/0bd7ea4a-c9ba-4a92-8943-9b0ed1cdb34c)

**Comment by jrieken** [comment]
I don't think this addresses my point of it being too busy, esp the misaligned frames/boxes on top of the background-coloured boxes are noisy.

**Comment by jrieken** [comment]
fyi @bpasero

**Comment by bpasero** [comment]
Related https://github.com/microsoft/vscode/issues/191833

**Comment by ThioJoe** [comment]
I think even being able to customize the background color of moved code would help a lot. Being able to distinguish moved code from truly added/removed lines would be a great help.

As far as I can tell, right now you can only set the color of the border with:
`diffEditor.move.border`, but not the background.

I'm imagining something like how the Notepad++ "Compare Plus" plugin works, you can see here moves are shown in blue, red is removals, green is additions. It makes it really easy to see moves vs actual changes. This even additions or removals within larger moved blocks.

<img width="1413" height="961" alt="Image" src="https://github.com/user-attachments/assets/937da821-7faf-4a79-b5db-ca5180ba2f80" />

--------

Looks like git even will work with additions/removals within moved blocks, making it even more useful to color move backgrounds:

<img width="1298" height="530" alt="Image" src="https://github.com/user-attachments/assets/773c7b8f-c487-4ba2-95ba-c3fb926f04bb" />
